File: Python_app/static/Credentials/Register/user_register.js
class User_register {
    constructor(firstname, lastname, email, password, confirm_password) {
        this._firstname = firstname;
        this._lastname = lastname;
        this._email = email;
        this._password = password;
        this._confirm_password = confirm_password;
    }


    get firstname() {
        return this._firstname;
    }

    set firstname(value) {
        this._firstname = value;
    }

    get lastname() {
        return this._lastname;
    }

    set lastname(value) {
        this._lastname = value;
    }

    get email() {
        return this._email;
    }

    set email(value) {
        this._email = value;
    }

    get password() {
        return this._password;
    }

    set password(value) {
        this._password = value;
    }

    get confirm_password() {
        return this._confirm_password;
    }

    set confirm_password(value) {
        this._confirm_password = value;
    }
}

export default User_register;

File: SpringBoot_app/databaseCore/src/main/java/com/example/databaseCore/Pojos/User/Req/UserRegister.java
package com.example.databaseCore.Pojos.User.Req;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

public class UserRegister {

    @NotBlank
    @Size(min = 1, max = 50)
    private String _firstname;

    @NotBlank
    @Size(min = 1, max = 50)
    private String _lastname;

    @NotBlank
    @Size(min = 5, max = 50)
    @Email
    private String _email;

    @NotBlank
    @Size(max = 100)
    @Pattern(regexp = "^(?=.*[A-Z].*[A-Z])(?=.*[!@#$&*])(?=.*[0-9].*[0-9])(?=.*[a-z].*[a-z].*[a-z]).{8,}$")
    private String _password;

    @NotBlank
    @Size(max = 100)
    private String _confirm_password;

    public UserRegister(String _firstname, String _lastname, String _email, String _password, String _confirm_password) {
        this._firstname = _firstname;
        this._lastname = _lastname;
        this._email = _email;
        this._password = _password;
        this._confirm_password = _confirm_password;
    }

    public String get_firstname() {
        return _firstname;
    }

    public void set_firstname(String _firstname) {
        this._firstname = _firstname;
    }

    public String get_lastname() {
        return _lastname;
    }

    public void set_lastname(String _lastname) {
        this._lastname = _lastname;
    }

    public String get_email() {
        return _email;
    }

    public void set_email(String _email) {
        this._email = _email;
    }

    public String get_password() {
        return _password;
    }

    public void set_password(String _password) {
        this._password = _password;
    }

    public String get_confirm_password() {
        return _confirm_password;
    }

    public void set_confirm_password(String _confirm_password) {
        this._confirm_password = _confirm_password;
    }
}
